src/services/hash.password.bcryptjs.ts

```diff
@@ -17,3 +20,28 @@ export async function hashPassword(
   const salt = await genSalt(rounds);
   return await hash(password, salt);
 }
+
+export interface PasswordHasher<T = string> {
+  hashPassword(password: T): Promise<T>;
+  comparePassword(provided_pass: T, stored_pass: T): Promise<boolean>;
```

Comment:
We use camelCase, not snake_case.

```suggestion
  comparePassword(providedPass: T, storedPass: T): Promise<boolean>;
```
```